src/Symfony/Component/Serializer/Normalizer/DenormalizerInterface.php

```diff
@@ -29,10 +29,16 @@ interface DenormalizerInterface
     /**
      * Denormalizes data back into an object of the given class.
      *
-     * @param mixed       $data    Data to restore
-     * @param string      $type    The expected class to instantiate
-     * @param string|null $format  Format the given data was extracted from
-     * @param array       $context Options available to the denormalizer
+     * @template TObject of object
+     *
+     * @param mixed                             $data    Data to restore
+     * @param class-string<TObject>|string|null $type    The expected class to instantiate
+     * @param string|null                       $format  Format the given data was extracted from
+     * @param array                             $context Options available to the denormalizer
+     *
+     * @phpstan-return ($type is class-string<TObject> ? TObject : mixed)
+     *
```

Comment:
```suggestion
```

PHPstan and Psalm read eachother vendor prefixed variants if they support it. We only use the `psalm-` vendor in Symfony code, as we're using Psalm in our CI.